packages/notification/core/notification.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  Renderable,
+  Notification,
+  NotificationOptions,
+  ValueOrFunction,
+} from '../types'
+import { genId } from './utils'
+import { dispatch, ActionType } from './store'
+
+type Message = ValueOrFunction<Renderable, Notification>
+
+type NotificationHandler = (
+  message: Message,
+  options?: NotificationOptions
+) => string
+
+const createNotification = (message: Message, options): Notification => ({
+  createdAt: Date.now(),
+  visible: true,
+  message,
+  pauseDuration: 0,
+  ...options,
+  id: options?.id || `notification-${genId()}`,
+})
+
+const createHandler = (): NotificationHandler => (message, options) => {
+  const notification = createNotification(message, options)
+  dispatch({ type: ActionType.UPSERT_NOTIFICATION, notification })
+  return notification.id
+}
+
+const notification = (message: Message, options?: NotificationOptions) => {
```

Comment:
I wonder if it would be possible for us to enforce usage of our `Notification*` components somehow, if that's something we want to do 🤔